osu.Game.Tests/Visual/Playlists/TestScenePlaylistsResultsScreen.cs

```diff
@@ -169,11 +190,13 @@ private void createResults(Func<ScoreInfo> getScore = null)
             });
 
             AddUntilStep("wait for screen to load", () => resultsScreen.IsLoaded);
-            waitForDisplay();
+            waitForDisplay(noScores);
         }
 
-        private void waitForDisplay()
+        private void waitForDisplay(bool noScores = false)
         {
+            if (noScores) return;
```

Comment:
This is quite baffling. `waitForDisplay(true)` does... nothing?

If this is the case, then rather than do this, please hoist out the `waitForDisplay()` call out of `createResults()`, and only call it in the tests that need it. This gets rid of the second `noScores` flag.